### ISSUE TITLE
(meta)names in squeeze() 

### ISSUE THREAD

**Issue by juangomezduaso** [DavisVaughan/rray#81]
I am strugling to understand the rationale behind the decision of not keeping dimnames in some cases (I am not sure in which ones) the way base drop() does.
Perhaps some concrete example of when, in a real situation, these names would be undesirable would help manuals readers baffled as I am.
Trying to figure it out I got this result that is perhaps caused by a bug in metadimname processing. If it is not I will really need help to get the point of this behavior

``` r

library(rray)
library(purrr)

#If I have this rray
( x<-rray(1:4, c(2,1,2), list(A=c("a1","a2"),B=c("b1"),C=c("c1","c2"))) )
#> <vctrs_rray<integer>[,1,2][4]>
#> , , C = c1
#> 
#>     B
#> A    b1
#>   a1  1
#>   a2  2
#> 
#> , , C = c2
#> 
#>     B
#> A    b1
#>   a1  3
#>   a2  4
# I get:
rray_squeeze(x)
#> <vctrs_rray<integer>[,2][4]>
#>     B
#> A    [,1] [,2]
#>   a1    1    3
#>   a2    2    4
```

<sup>Created on 2019-04-11 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by DavisVaughan** [comment]
Yea, you're right it doesn't do the right think. I wasn't thinking about this right when I made the function. Here is what should happen:

``` r
library(rray)
#> Registered S3 method overwritten by 'vctrs':
#>   method                   from 
#>   as.character.rlang_error rlang

# (2, 1, 2)
x <- rray(
  1:4, 
  c(2,1,2), 
  list(
    A = c("a1", "a2"),
    B = c("b1"),
    C = c("c1", "c2")
  )
)

# Go from:
# (2, 1, 2) -> (2, 2)
# Should:
# Drop all names from dim 2
# Move all names from dim 3 to dim 2
rray_squeeze(x, axes = 2)
#> <vctrs_rray<integer>[,2][4]>
#>     B
#> A    [,1] [,2]
#>   a1    1    3
#>   a2    2    4

# essentially should do:
drop(x)
#> <vctrs_rray<integer>[,2][4]>
#>     C
#> A    c1 c2
#>   a1  1  3
#>   a2  2  4


# another example from the help docs
y <- rray(1:10, c(10, 1))
y <- set_row_names(y, letters[1:10])
y <- t(y)

y
#> <vctrs_rray<integer>[,10][10]>
#>      a b c d e f g h i  j
#> [1,] 1 2 3 4 5 6 7 8 9 10

# going from:
# (1, 10) -> (10)
# really should drop names from 1st dimension
# move names from 2nd dim to 1st dim
rray_squeeze(y)
#> <vctrs_rray<integer>[10]>
#>  [1]  1  2  3  4  5  6  7  8  9 10

# should do this
drop(as.matrix(y))
#>  a  b  c  d  e  f  g  h  i  j 
#>  1  2  3  4  5  6  7  8  9 10
```

<sup>Created on 2019-04-13 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by juangomezduaso** [comment]
Exactly. This is the behavior that makes sense to me and also is consistent with what  rray_expand_dims() does inversely

**Comment by DavisVaughan** [comment]
Closed by #85
